pkg/cloud/metadata_test.go

```diff
@@ -325,11 +327,22 @@ func TestNewMetadataService(t *testing.T) {
 			getInstanceDocErr: nil,
 			getOutpostArnErr:  fmt.Errorf("405"),
 		},
+		{
+			name:            "fail: kube client is nil",
+			isAvailable:     false,
+			isClientSetNil:  true,
+			clientSetNilErr: fmt.Errorf("instance metadata is unavailable and kubernetes clientset is nil"),
+		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			clientset := fake.NewSimpleClientset(&tc.node)
+			var clientset *fake.Clientset
```

Comment:
What am I doing wrong here? For some reason I can't get this test case to pass

Reply:
can you paste the error?

Reply:
```
=== RUN   TestNewMetadataService/fail:_kube_client_is_nil
W0520 18:25:16.684313   23724 metadata.go:107] EC2 instance metadata is not available
    metadata_test.go:412: NewMetadataService() returned an unexpected error. Expected instance metadata is unavailable and kubernetes clientset is nil, got instance metadata is unavailable and CSI_NODE_NAME env var not set
```

Reply:
so even though you've set clientset to nil, it's not hitting the if condition you added to `NewMetadataService`, is that right?

Reply:
what dictates whether mockEC2Metadata.Available returns true or not?

Reply:
I think because the pointer got cast into an interface. the interface is not nil even if the implementation is nil.